test-data/unit/check-based-misc.test

```diff
@@ -21,3 +21,13 @@ if True:
     1  # type: ignore[operator]
 [out]
 main:3:5:3:31: error: Unused "type: ignore" comment  [unused-ignore]
+
+
+[case testUsingDecoratedNone]
+from typing import TypeVar
+T = TypeVar("T")
+def deco(t: T) -> T: ...
```

Comment:
Tests for decorators that change the return type (both to `None` and from `None` to something else)